### PR TITLE
SN-12/Create Profile model

### PR DESCRIPTION
**Pull request by Slavahaim**
Create Profile model
Add validates

### CODE REVIEW

File: app/models/profile.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Profile < ApplicationRecord
+  #has_one_attached :photo
+  #belongs_to :user
+  #has_many :user_interests
+
+  validates :surname, presence: true, length: { maximum: 255 }
+  validates :name, presence: true, length: { maximum: 255 }
+  validates :patronymic, length: { maximum: 255 }
+  validates :birthday, presence: true
+  validates :email, presence: true, uniqueness: true, length: { maximum: 255 }
```
Comment (ramantechart) [inline app/models/profile.rb]: Email size limit is around 20Mb.

File: app/models/profile.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  validates :phone, presence: true, uniqueness: true, length: { maximum: 20 }
```
Comment (ramantechart) [inline app/models/profile.rb]: 15 digits max
https://en.wikipedia.org/wiki/Telephone_numbering_plan

File: config/cable.yml.example [hunk 3]
```diff
@@ -1,10 +1,10 @@
-development:
```
Comment (ramantechart) [inline config/cable.yml.example]: this change doesn't make any sense

File: config/database.yml.example [hunk 4]
```diff
@@ -1,88 +1,88 @@
-# PostgreSQL. Versions 9.3 and up are supported.
-#
-# Install the pg driver:
-#   gem install pg
-# On macOS with Homebrew:
-#   gem install pg -- --with-pg-config=/usr/local/bin/pg_config
```
Comment (ramantechart) [inline config/database.yml.example]: this change doesn't make any sense

File: config/storage.yml.example [hunk 5]
```diff
@@ -1,34 +1,34 @@
-test:
```
Comment (ramantechart) [inline config/storage.yml.example]: this change doesn't make any sense

File: db/migrate/20220120124837_create_profiles.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+class CreateProfiles < ActiveRecord::Migration[6.1]
+  def change
+    create_table :profiles do |t|
+      t.string :surname, null: false
```
Comment (ramantechart) [inline db/migrate/20220120124837_create_profiles.rb:7]: we are missing validations such as string size.

File: db/seeds.rb [hunk 7]
```diff
@@ -5,3 +5,15 @@
...
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+require 'faker'
+
+require 'factory_bot_rails'
+
+GENERATE_PROFILE_COUNT = 30
+
+ActiveRecord::Base.transaction do
```
Comment (ramantechart) [inline db/seeds.rb]: FactoryBot.create_list(:profile, GENERATE_PROFILE_COUNT)

File: db/seeds.rb [hunk 8]
```diff
@@ -5,3 +5,15 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+require 'faker'
```
Comment (ramantechart) [inline db/seeds.rb]: why?
you don't need it

File: spec/factories/profiles.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+FactoryBot.define do
+  factory :profile do
+    surname { Faker::Name.last_name }
+    name { Faker::Name.first_name }
+    patronymic { Faker::Name.middle_name }
+    birthday { Faker::Date.birthday(min_age: 18, max_age: 65) }
+    email { Faker::Internet.unique.email }
+    phone { Faker::PhoneNumber.unique.cell_phone_in_e164 }
+    #photo { nil }
```
Comment (ramantechart) [inline spec/factories/profiles.rb]: it's not your problem.
Liza will add photo.
You can remove this

File: spec/factories/profiles.rb [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    about { "#{Faker::Address.full_address}, #{Faker::University.name}, #{Faker::Hobby.activity}"  }
+    hidden { false }
+    verified { false }
```
Comment (ramantechart) [inline spec/factories/profiles.rb:35]: we should also have traits `hidden` and `verifieed`

File: spec/models/profile_spec.rb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Profile, type: :model do
+    pending "add some examples to (or delete) #{__FILE__}"
```
Comment (ramantechart) [inline spec/models/profile_spec.rb]: Missing specs

File: spec/spec_helper.rb [hunk 12]
```diff
@@ -44,53 +44,51 @@
   # triggering implicit auto-inclusion in groups with matching metadata.
   config.shared_context_metadata_behavior = :apply_to_host_groups
 
-# The settings below are suggested to provide a good initial experience
```
Comment (ramantechart) [inline spec/spec_helper.rb]: this change doesn't make any sense

**Review by ramantechart — CHANGES_REQUESTED** [review]

**Review by skripchenkoVeta — APPROVED** [review]

File: db/seeds.rb (hunk 8 above)
Comment (Slavahaim) [reply db/seeds.rb]: remove

**Review by PivovarAndrew — APPROVED** [review]

**Review by PFDeve — APPROVED** [review]
Please remove the changes which Roman said `this change doesn't make any sense` before merge

File: config/cable.yml.example (hunk 3 above)
Comment (Slavahaim) [reply config/cable.yml.example]: how i can fix it?
I didn't change these files.
When I set up my local database, I deleted them. Then added it back. But the git decided to comment them out as modified.

File: config/database.yml.example (hunk 4 above)
Comment (Slavahaim) [reply config/database.yml.example]: how i can fix it?
I didn't change these files.
When I set up my local database, I deleted them. Then added it back. But the git decided to comment them out as modified.

File: config/storage.yml.example (hunk 5 above)
Comment (Slavahaim) [reply config/storage.yml.example]: how I can fix it?
I didn't change these files.
When I set up my local database, I deleted them. Then added it back. But the git decided to comment them out as modified.

File: spec/factories/profiles.rb [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    name { Faker::Name.first_name }
+    patronymic { Faker::Name.middle_name }
+    birthday { Faker::Date.birthday(min_age: 18, max_age: 65) }
+    email { Faker::Internet.unique.email }
+    phone { Faker::PhoneNumber.unique.cell_phone_in_e164 }
+    about { "#{Faker::Address.full_address}, #{Faker::University.name}, #{Faker::Hobby.activity}" }
+    hidden { false }
+    verified { false }
+
+    trait(:hidden_true) { hidden { true } }
```
Comment (mlshvch) [inline spec/factories/profiles.rb:37]: I suggest to rename to `:hidden`

File: spec/factories/profiles.rb [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    trait(:verified_true) { verified { true } }
```
Comment (mlshvch) [inline spec/factories/profiles.rb:38]: I suggest to rename to `:verified`

File: spec/models/profile_spec.rb [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+  end
+
+  describe 'phone limit test' do
+    context 'when phone limit not more than 15 digits' do
+      it 'ensures phone limit <= 15' do
+        expect(profile.phone.size <= 15).to eq(true)
+      end
+
+      it 'return error' do
+        profile.phone = '1234567890123456'
```
Comment (mlshvch) [inline spec/models/profile_spec.rb:75]: To my mind, it's better to use Faker for generating such things

**Review by mlshvch — APPROVED** [review]

**Review by ramantechart — APPROVED** [review]

**Pull request merged**